sw/airborne/modules/rot_wing_drone/rot_wing_automation.c

```diff
@@ -0,0 +1,200 @@
+#define ROT_WING_AUTOMATION_TRANS_LENGTH 200.0
+#endif
+
+#ifndef ROT_WING_AUTOMATION_TRANS_AIRSPEED
+#define ROT_WING_AUTOMATION_TRANS_AIRSPEED 15.0
+#endif
+
+#ifndef ROT_WING_AUTOMATION_BASE_DIST
+#define ROT_WING_AUTOMATION_BASE_DIST 100.
+#endif
+
+#ifndef ROT_WING_AUTOMATION_WIND_FILT_CUTOFF
+#define ROT_WING_AUTOMATION_WIND_FILT_CUTOFF 0.001
+#endif
+
+struct rot_wing_automation rot_wing_a;
+
+// Declare filters (for windspeed estimation)
+Butterworth2LowPass rot_wing_automation_wind_filter[2]; // Wind filter
+
+// declare function
+inline void update_waypoint_rot_wing_automation(uint8_t wp_id, struct FloatVect3 * target_ned);
+inline void update_wind_vector(void);
+
+void init_rot_wing_automation(void)
+{
+  rot_wing_a.trans_accel = ROT_WING_AUTOMATION_TRANS_ACCEL;
+  rot_wing_a.trans_decel = ROT_WING_AUTOMATION_TRANS_DECEL;
+  rot_wing_a.trans_length = ROT_WING_AUTOMATION_TRANS_LENGTH;
+  rot_wing_a.trans_airspeed = ROT_WING_AUTOMATION_TRANS_AIRSPEED;
+
+  rot_wing_a.transitioned = false;
+  rot_wing_a.windvect.x = 0.0;
+  rot_wing_a.windvect.y = 0.0;
+  rot_wing_a.windvect_f.x = 0.0;
+  rot_wing_a.windvect_f.y = 0.0;
+
+  // Init windvector low pass filter
+  float tau = 1.0 / (2.0 * M_PI * ROT_WING_AUTOMATION_WIND_FILT_CUTOFF);
+  float sample_time = 1.0 / 10.0;
+  init_butterworth_2_low_pass(&rot_wing_automation_wind_filter[0], tau, sample_time, 0.0);
+  init_butterworth_2_low_pass(&rot_wing_automation_wind_filter[1], tau, sample_time, 0.0);
+}
+
+// periodic function
+void periodic_rot_wing_automation(void)
+{
+  update_wind_vector();
+  float airspeed = stateGetAirspeed_f();
+  if (airspeed > rot_wing_a.trans_airspeed)
+  {
+    rot_wing_a.transitioned = true;
+  } else {
+    rot_wing_a.transitioned = false;
+  }
+}
+
+// Update a waypoint such that you can see on the GCS where the drone wants to go
+void update_waypoint_rot_wing_automation(uint8_t wp_id, struct FloatVect3 * target_ned) {
+
+  // Update the waypoint
+  struct EnuCoor_f target_enu;
+  ENU_OF_TO_NED(target_enu, *target_ned);
+  waypoint_set_enu(wp_id, &target_enu);
+
+  // Send waypoint update every half second
+  RunOnceEvery(100/2, {
+    // Send to the GCS that the waypoint has been moved
+    DOWNLINK_SEND_WP_MOVED_ENU(DefaultChannel, DefaultDevice, &wp_id,
+                                &waypoints[wp_id].enu_i.x,
+                                &waypoints[wp_id].enu_i.y,
+                                &waypoints[wp_id].enu_i.z);
+  } );
+}
+
+void update_wind_vector(void)
+{
+  float psi = stateGetNedToBodyEulers_f()->psi;
+
+  float cpsi = cosf(psi);
+  float spsi = sinf(psi);
+
+  float airspeed = stateGetAirspeed_f();
+  struct NedCoor_f *groundspeed = stateGetSpeedNed_f();
+  struct FloatVect2 airspeed_v = { cpsi * airspeed, spsi * airspeed };
+  VECT2_DIFF(rot_wing_a.windvect, *groundspeed, airspeed_v);
+
+  // Filter the wind
+  update_butterworth_2_low_pass(&rot_wing_automation_wind_filter[0], rot_wing_a.windvect.x);
```

Comment:
`rot_wing_a.windvect_f.x =  update_butterworth_2_low_pass(&rot_wing_automation_wind_filter[0], rot_wing_a.windvect.x);`
Is nicer and works also

Reply:
good idea!